sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -50,6 +50,9 @@ object CommandUtils extends Logging {
       catalog.alterTableStats(table.identifier, Some(newStats))
     } else if (table.stats.nonEmpty) {
       catalog.alterTableStats(table.identifier, None)
+    } else {
+      // In other cases, we still need to invalidate the table relation cache.
```

Comment:
to confirm: does `catalog.alterTableStats` refresh the relation cache?

Reply:
Yes.
https://github.com/apache/spark/blob/1743d5be7f74a5a7ab28dbc5c0f431cf2c2dfbc0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L409-L422